stingray/pulse/overlapandsave/ols.py

```diff
@@ -0,0 +1,355 @@
+
+    `a` must be greater than 1, and `x` must be greater than 0.
+    # Examples
+    ```jldoctest
+    julia> nextpow(2, 7)
+    8
+    julia> nextpow(2, 9)
+    16
+    julia> nextpow(5, 20)
+    25
+    julia> nextpow(4, 16)
+    16
+    ```
+    """
+    assert x > 0 and a > 1
+    if x <= 1:
+        return 1.0
+    n = math.ceil(math.log(x, a))
+    p = a ** (n - 1)
+    return p if p >= x else a ** n
+
+
+def nextprod(a: List[int], x: int) -> int:
+    """Next integer greater than or equal to `x` that can be written as ``\\prod k_i^{a_i}`` for integers
+    ``a_1``, ``a_2``, etc.
+    # Examples
+    ```jldoctest
+    julia> nextprod([2, 3], 105)
+    108
+    julia> 2^2 * 3^3
+    108
+    ```
+    """
+    k = len(a)
+    v = [1] * k  # current value of each counter
+    mx = [nextpow(ai, x) for ai in a]  # maximum value of each counter
+    v[0] = mx[0]  # start at first case that is >= x
+    p = mx[0]  # initial value of product in this case
+    best = p
+    icarry = 1
+
+    while v[-1] < mx[-1]:
+        if p >= x:
+            best = p if p < best else best  # keep the best found yet
+            carrytest = True
+            while carrytest:
+                p = p // v[icarry - 1]
+                v[icarry - 1] = 1
+                icarry += 1
+                p *= a[icarry - 1]
+                v[icarry - 1] *= a[icarry - 1]
+                carrytest = v[icarry - 1] > mx[icarry - 1] and icarry < k
+            if p < x:
+                icarry = 1
+        else:
+            while p < x:
+                p *= a[0]
+                v[0] *= a[0]
+    return int(mx[-1] if mx[-1] < best else best)
+
+
+def array_range(start: List[int], stop: List[int], step: List[int]) -> \
+        Iterable[Tuple]:
+    """
+    Makes an iterable of non-overlapping slices, e.g., to partition an array
+
+    Returns an iterable of tuples of slices, each of which can be used to
+    index into a multidimensional array such as Numpy's ndarray.
+
+    >> [arr[tup] for tup in array_range([0, 0], arr.shape, [5, 7])]
+
+    where `arr` can be indexed with a tuple of slices (e.g., Numpy), will
+    evaluate to a list of sub-arrays.
+
+    Same arguments as `range` except all three arguments are required and
+    expected to be list-like of same length. `start` indicates the indexes
+    to start each dimension. `stop` indicates the stop index for each
+    dimension. `step` is the size of the chunk in each dimension.
+    """
+    assert len(start) == len(stop)
+    assert len(stop) == len(step)
+    assert all(map(lambda x: x > 0, step))
+    startRangesGen = map(lambda v: range(*v), zip(start, stop, step))
+    startToSliceMapper = lambda multiStart: tuple(
+        slice(i, min(i + step, stop)) for i, stop, step in
+        zip(multiStart, stop, step))
+    return map(startToSliceMapper, product(*startRangesGen))
+
+
+def prepareh(h, nfft: List[int], rfftn=None):
+    """Pre-process a filter array
```

Comment:
As above: some details about the input/output variables in the standard format would be useful.